plugin/metadata/metadataer.go

```diff
@@ -0,0 +1,53 @@
+package metadata
+
+import (
+	"context"
+
+	"github.com/miekg/dns"
+)
+
+// Metadataer interface needs to be implemented by each plugin willing to provide
+// metadata information for other plugins.
+// Note: this method should work quickly, because it is called for every request
+// from the metadata plugin.
+type Metadataer interface {
```

Comment:
Since it is no longer a single function interface, can we change the interface name. Maybe "MetadataProvider"?